src/openeo_test_suite/tests/processes/metadata/test_process_metadata.py

```diff
@@ -0,0 +1,142 @@
+import json
+import pytest
+import requests
+from openeo_test_suite.lib.process_selection import get_selected_processes
+from openeo_test_suite.lib.backend_under_test import (
+    get_backend_under_test,
+)
+
+
+@pytest.fixture(scope="module")
+def api_processes():
+    endpoint_path = "processes"
+    base_url = get_backend_under_test().connection.root_url
+    if not base_url:
+        raise ValueError("No backend URL configured")
+    if base_url.endswith("/"):
+        base_url = base_url[:-1]
+    full_endpoint_url = f"{base_url}/{endpoint_path}"
+    response = requests.get(full_endpoint_url)
+    if response.status_code != 200:
+        raise ValueError(
+            f"Failed to get processes from {full_endpoint_url}: {response.content}"
+        )
+    return json.loads(response.content)["processes"]
+
+
+_get_selected_processes = get_selected_processes.__wrapped__
+
+
+def get_examples():
+    """Collect process examples/tests from examples root folder containing JSON5 files."""
+    return [process for process in _get_selected_processes()]
+
+
+@pytest.mark.parametrize(
+    "expected_process",
+    [process for process in get_examples() if process.metadata != {}],
+    ids=[process.process_id for process in get_examples() if process.metadata != {}],
+)
```

Comment:
related to the above discussion about `__wrapped__`, I think the whole parameterization setup can be simplified.

instead of `_get_selected_processes` , `def get_examples()...` and the current `@pytest.mark.parametrize(`
you can do something like this:

```python
def _get_test_id(val):
    if isinstance(val, ProcessData):
        return val.process_id


@pytest.mark.parametrize(
    "expected_process",
    [p for p in get_selected_processes() if p.metadata],
    ids=_get_test_id
)
def test_process_metadata_functional(api_processes, expected_process, skipper):
```

which eliminates some redirection and boilerplate

Reply:
I've implemented this, in the newest commits.